ipykernel/ipkernel.py

```diff
@@ -261,6 +270,42 @@ def do_complete(self, code, cursor_pos):
                 'metadata' : {},
                 'status' : 'ok'}
 
+    def _experimental_do_complete(self, code, cursor_pos):
+        """
+        Experimental completions from IPython, using Jedi. 
+        """
+        if cursor_pos is None:
+            cursor_pos = len(code)
+        with provisionalcompleter():
+            raw_completions = self.shell.Completer.completions(code, cursor_pos)
+            completions = list(rectify_completions(code, raw_completions))
+            
+            comps = []
+            for comp in completions:
+                comps.append(dict(
+                            start=comp.start,
+                            end=comp.end,
+                            text=comp.text,
+                            type=comp.type,
+                              ))
+
+        if completions:
+            s = completions[0].start
```

Comment:
Seems reasonable to use `completions[0]` but might there ever be a reason to not pick this one?